pilot/proxy/envoy/config.go

```diff
@@ -326,12 +317,15 @@ func buildRDSRoute(mesh *meshconfig.MeshConfig, node proxy.Node, routeName strin
 		return nil, err
 	}
 
+	if httpConfigs[port] == nil {
```

Comment:
I assume this is a bug fix unrelated to the main change ? Same for buildIngressRoutes - is it just a refactoring, or does it fixes something ?

Reply:
This is part of the fix. This will make sure that if err != nil, a valid result (although empty in this case) will be returned, versus a nil pointer returned.